e2e-tests/portal/cypress/support/portal/createBssEwcsDeal.js

```diff
@@ -0,0 +1,178 @@
+  cy.login(BANK1_MAKER1);
+
+  dashboard.createNewSubmission().click();
+
+  // select BSS scheme
+  selectScheme.bss().click();
+  cy.clickContinueButton();
+
+  // select True before starting the application
+  beforeYouStart.true().click();
+  cy.clickSubmitButton();
+
+  // complete "bank details"
+  cy.keyboardInput(bankDetails.bankDealId(), '123');
+  cy.keyboardInput(bankDetails.bankDealName(), 'BssDeal');
+
+  cy.clickSubmitButton();
+
+  if (readyForCheck) {
+    contract.aboutSupplierDetailsLink().click();
+
+    if (exporterCompanyName) {
+      contractAboutSupplier.supplierType().select('Exporter');
+      cy.keyboardInput(contractAboutSupplier.supplierName(), exporterCompanyName);
+      contractAboutSupplier.supplierAddress().country().select('United Kingdom');
+      cy.keyboardInput(contractAboutSupplier.supplierAddress().line1(), 'Test');
+      cy.keyboardInput(contractAboutSupplier.supplierPostCode(), 'AB1 2CD');
+      contractAboutSupplier.supplierCorrespondenceAddressSame().click();
+      contractAboutSupplier.industrySector().select('Accommodation and food service activities');
+      contractAboutSupplier.industryClass().select('Event catering activities');
+      contractAboutSupplier.smeTypeSmall().click();
+      cy.keyboardInput(contractAboutSupplier.supplyContractDescription(), 'Supply Contract Description');
+      contractAboutSupplier.notLegallyDistinct().click();
+      contractAboutSupplier.nextPage().click();
+    } else {
+      // complete "about supplier"
+      contractAboutSupplier.supplierType().select(submissionDetails['supplier-type']);
+      cy.keyboardInput(contractAboutSupplier.supplierCompaniesHouseRegistrationNumber(), '12345678');
+      contractAboutSupplier.supplierSearchCompaniesHouse().click();
+      contractAboutSupplier.supplierAddress().country().select('United Kingdom');
+      contractAboutSupplier.supplierCorrespondenceAddressSame().click();
+      contractAboutSupplier.smeTypeSmall().click();
+      cy.keyboardInput(contractAboutSupplier.supplyContractDescription(), 'Supply Contract Description');
+      contractAboutSupplier.notLegallyDistinct().click();
+      contractAboutSupplier.nextPage().click();
+    }
+
+    // complete "about buyer"
+    cy.keyboardInput(contractAboutBuyer.buyerName(), 'Buyer Name');
+    contractAboutBuyer.buyerAddress().country().select('United Kingdom');
+    cy.keyboardInput(contractAboutBuyer.buyerAddress().line1(), 'Line 1');
+    cy.keyboardInput(contractAboutBuyer.buyerAddress().line2(), 'Line 2');
+    cy.keyboardInput(contractAboutBuyer.buyerAddress().line3(), 'Line 3');
+    cy.keyboardInput(contractAboutBuyer.buyerAddress().town(), 'Town');
+    cy.keyboardInput(contractAboutBuyer.buyerAddress().postcode(), 'AB1 2CD');
+    contractAboutBuyer.destinationOfGoodsAndServices().select('United Kingdom');
+    contractAboutBuyer.nextPage().click();
```

Comment:
rather than using comments to separate out these sections it would be nicer & self documenting to break the function up

```
const createBssEwcsDeal = ({ readyForCheck = false, dealType, facilityStage, exporterCompanyName }) => {
    cy.login(BANK1_MAKER1);

  dashboard.createNewSubmission().click();

  // select BSS scheme
  selectScheme.bss().click();
  cy.clickContinueButton();

  // select True before starting the application
  beforeYouStart.true().click();
  cy.clickSubmitButton();

  // complete "bank details"
  cy.keyboardInput(bankDetails.bankDealId(), '123');
  cy.keyboardInput(bankDetails.bankDealName(), 'BssDeal');

  cy.clickSubmitButton();

  completeExporterDetails(exporterCompanyName);

  completeAboutBuyer();

  completeFinancialDetails();

  completeEligibilityCriteria(dealSubmissionType);

  addBond(facilityStage);

  // Proceed to review button
  contract.proceedToReview().should('exist');
  contract.proceedToReview().click();

  // submit to checker
  cy.keyboardInput(contractReadyForReview.comments(), 'Ready for checkers approval');
  contractReadyForReview.readyForCheckersApproval().click();
};
```

Reply:
Done